### PR TITLE
dotCMS/core#15135 Move all menu state to the dot-navigation.service

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/app/view/components/dot-navigation/services/dot-navigation.service.ts [hunk 1]
```diff
@@ -58,16 +86,57 @@ export class DotNavigationService {
...
+        this.dotRouterService.gotoPortlet(url);
+    }
+
     /**
      * Navigates to the first portlet
      *
      * @memberof DotNavigationService
      */
-    goToFirstPortlet(): Promise<boolean> {
+    goToFirstPortlet(): Promise < boolean > {
```
Comment (alfredo-dotcms) [inline src/app/view/components/dot-navigation/services/dot-navigation.service.ts]: remove space

File: src/app/view/components/dot-navigation/services/dot-navigation.service.ts [hunk 2]
```diff
@@ -77,24 +146,13 @@ export class DotNavigationService {
...
-    }
-
     /**
      * Emit event when navigation end
      *
      * @returns {Observable<Event>}
      * @memberof DotNavigationService
      */
-    onNavigationEnd(): Observable<Event> {
+    onNavigationEnd(): Observable < Event > {
```
Comment (alfredo-dotcms) [inline src/app/view/components/dot-navigation/services/dot-navigation.service.ts]: remove space

File: src/app/view/components/dot-navigation/dot-navigation.component.spec.ts [hunk 3]
```diff
@@ -89,15 +34,48 @@ class HostTestComponent {
...
-        return of([dotMenuMock(), dotMenuMock1()]);
+        return this._items$.asObservable();
     }
 
     onNavigationEnd(): Observable<NavigationEnd> {
         return this._routeEvents.asObservable().pipe(skip(1));
     }
 
+    setOpen() {
+        console.log('setOpen');
```
Comment (hmoreras) [inline src/app/view/components/dot-navigation/dot-navigation.component.spec.ts]: remove ?

File: src/app/view/components/dot-navigation/dot-navigation.component.spec.ts [hunk 4]
```diff
@@ -89,15 +34,48 @@ class HostTestComponent {
...
+            {
+                ...dotMenuMock(),
+                isOpen: true,
+            },
+            dotMenuMock1()
+        ]);
+    }
+
+    expandMenu() {
+        console.log('expandMenu');
```
Comment (hmoreras) [inline src/app/view/components/dot-navigation/dot-navigation.component.spec.ts]: remove ?

**Pull request merged**